src/posix/platform/shared_lib.cpp

```diff
@@ -0,0 +1,272 @@
+}
+
+void *pThreadOtMainLoop(void *arg)
+{
+    pthread_mutex_lock(&gLock);
+    otPlatformConfig platformConfig;
+    int              interfaceIdx                = 0;
+    char             radioUrl[MAX_RADIO_URL_LEN] = {0};
+    char             iface[MAX_INTERFACE_LEN]    = {0};
+    Param           *initParam                   = (Param *)arg;
+
+    syslog(LOG_INFO, "Inside pThreadOtMainLoop");
+
+    interfaceIdx = getInterface();
+    sprintf(iface, "wpan%d", interfaceIdx);
+    syslog(LOG_INFO, "interface found [%s]", iface);
+
+    if (!getRadioURL(initParam->comPort))
+    {
+        syslog(LOG_CRIT, "radio device not found");
+        return NULL;
+    }
+    sprintf(radioUrl, "spinel+hdlc+uart:///dev/%s", initParam->comPort);
+    syslog(LOG_INFO, "radio Url found [%s]", radioUrl);
+    syslog(LOG_INFO, "debug level [%d]", initParam->debugLevel);
+
+    platformConfig.mInterfaceName = iface;
+    VerifyOrDie(platformConfig.mRadioUrlNum < OT_ARRAY_LENGTH(platformConfig.mRadioUrls), OT_EXIT_INVALID_ARGUMENTS);
+    platformConfig.mRadioUrls[platformConfig.mRadioUrlNum++] = radioUrl;
+#if defined(__linux)
+    platformConfig.mRealTimeSignal = 41;
+#endif
+    platformConfig.mSpeedUpFactor = 1;
+
+    syslog(LOG_INFO, "Running %s", otGetVersionString());
+    syslog(LOG_INFO, "Thread version: %hu", otThreadGetVersion());
+    IgnoreError(otLoggingSetLevel(initParam->debugLevel));
+
+    gInstance = otSysInit(&platformConfig);
+    VerifyOrDie(gInstance != NULL, OT_EXIT_FAILURE);
+    syslog(LOG_INFO, "Thread interface: %s", otSysGetThreadNetifName());
+    syslog(LOG_INFO, "RCP version: %s", otPlatRadioGetVersionString(gInstance));
+
+    if (platformConfig.mDryRun)
+    {
+        exit(OT_EXIT_SUCCESS);
+    }
+
+    otLogInfoPlat("ot instance create success!!!");
+
+    free(initParam);
+
+    pthread_mutex_unlock(&gLock);
+
+    while (gTerminate == false)
+    {
+        otSysMainloopContext mainloop;
+
+        otTaskletsProcess(gInstance);
+
+        FD_ZERO(&mainloop.mReadFdSet);
+        FD_ZERO(&mainloop.mWriteFdSet);
+        FD_ZERO(&mainloop.mErrorFdSet);
+
+        mainloop.mMaxFd           = -1;
+        mainloop.mTimeout.tv_sec  = 10;
+        mainloop.mTimeout.tv_usec = 0;
+
+        otSysMainloopUpdate(gInstance, &mainloop);
+
+        if (otSysMainloopPoll(&mainloop) >= 0)
+        {
+            pthread_mutex_lock(&gLock);
+
+            otSysMainloopProcess(gInstance, &mainloop);
+
+            pthread_mutex_unlock(&gLock);
+        }
+        else if (errno != EINTR)
+        {
+            perror("select");
+            ExitNow();
+        }
+    }
+
+exit:
+    otSysDeinit();
+    gThreadId  = 0;
+    gInstance  = NULL;
+    gTerminate = false;
+    syslog(LOG_INFO, "terminate thread mainloop : exit");
+    return NULL;
+}
+
+void otSysGetInstance(otInstance **instance, const char *comPort, uint16_t debug)
+{
+    Param *initParam = (Param *)malloc(sizeof(Param));
+    strcpy(initParam->comPort, comPort);
```

Comment:
Use `strncpy()` to be safe and avoid any overwrite over `initParm->comPort` buffer size?